### PR TITLE
Put clang plugins in usr/local/lib

### PR DESCRIPTION
**Pull request by pr0t4zy**

### CODE REVIEW

File: geticeccversion.sh [hunk 1]
```diff
@@ -46,26 +47,40 @@ MAC_ENV_PATH="${ICECC_ENV_DIR}/${ENV_NAME}"
...
-  TEMP_ENV_PATH=${ICECC_ENV_DIR}/${TEMP_ENV_FILENAME}
+  TEMP_ENV_PATH=${ICECC_ENV_TEMP_DIR}/${TEMP_ENV_FILENAME}
   if [ ! -e "$TEMP_ENV_PATH" ]; then
     echo "Error: Can't find environment created at $TEMP_ENV_PATH." >&2
     exit 1
   fi
 
-  mv "$TEMP_ENV_PATH" "$MAC_ENV_PATH"
+  pushd $ICECC_ENV_TEMP_DIR
+  tar xzf $TEMP_ENV_FILENAME 1>/dev/null
```
Comment (avl7771) [inline geticeccversion.sh]: There's no need to untar and retar the whole file - as long as it's not compressed you can add files to the archive using the 'r' option. Then you won't need a temporary directory to copy the files to before adding them to the archive.

File: create-icecc-env-linux.py [hunk 2]
```diff
@@ -71,6 +71,16 @@ def DownloadUrl(url, output_path):
         time.sleep(retry_wait_s)
         retry_wait_s *= 2
 
+def MovePlugin(plugin_base_name, clang_prefix_path):
+    plugin_src_name = plugin_base_name + ".so"
```
Comment (avl7771) [inline create-icecc-env-linux.py:75]: If we want this to work correctly on Linux as well (eventually) should we add it both as .so and as .dylib to the package?

**Review by avl7771 — CHANGES_REQUESTED** [review]

File: create-icecc-env-linux.py (hunk 2 above)
Comment (pr0t4zy) [reply create-icecc-env-linux.py:75]: I did not analyze yet what would be required to make it work on Linux, I suspect that there will be more to it then just putting "so" in the package. I think this should be part of "Making it work on Linux" change.

Comment (avl7771) [reply create-icecc-env-linux.py:75]: OK.

File: geticeccversion.sh [hunk 3]
```diff
@@ -56,31 +55,38 @@ if [ ! -e "$MAC_ENV_PATH" ]; then
...
-  mkdir -p usr/local/lib
-  cp $PLUGINS_PATH/libFindBadConstructs.dylib usr/local/lib/
-  cp $PLUGINS_PATH/libBlinkGCPlugin.dylib usr/local/lib/
-  rm $TEMP_ENV_FILENAME
-  tar czf $MAC_ENV_PATH . 1>/dev/null
-  popd
 
-  rm -rf $ICECC_ENV_TEMP_DIR
+  gunzip $ENV_NAME
+  cp $PLUGINS_PATH/libFindBadConstructs.dylib $PLUGIN_DST_DIR/
```
Comment (avl7771) [inline geticeccversion.sh]: Instead of copying the file, we could do a symlink and tar with the '-L' option.

**Review by avl7771 — CHANGES_REQUESTED** [review]

File: geticeccversion.sh (hunk 3 above)
Comment (pr0t4zy) [reply geticeccversion.sh]: Apparently the link target path is too long, I get: "Link contents too long". We hit USTAR_linkname_size which is 100.

Comment (pr0t4zy) [reply geticeccversion.sh]: OK, I got the order wrong.

**Review by avl7771 — APPROVED** [review]

**Pull request merged**